File: src/main/java/com/example/appjambackend/global/error/exception/ErrorCode.java
package com.example.appjambackend.global.error.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ErrorCode {

    INVALID_TOKEN(401, "Invalid Token"),
    EXPIRED_TOKEN(401, "Expired Token"),

    USER_NOT_FOUND(404, "User Not Found"),

    FEED_NOT_FOUND_EXCEPTION(404, "feed Not Found"),

    FILE_UPLOAD_FAIL(400,"File Upload Fail"),
    FILE_EMPTY(400, "File is Empty"),

    TOO_MANY_FILES(413,"TOO Many Files"),
    USERID_ALREADY_EXISTS(409, "Userid Already Exists"),
    FEED_NOT_FOUND_EXCEPTION(404, "feed Not Found")
    ;

    private final int statusCode;
    private final String message;
}

File: mitalk/src/main/kotlin/com/example/mitalk/global/exception/ErrorCode.kt
package com.example.mitalk.global.exception

enum class ErrorCode(
        val message: String,
        val status: Int
) {
    FILE_SIZE_OVER("파일의 크기가 1024MB보다 큼", 400),

    UNAUTHORIZED("승인되지 않았습니다.", 401),
    EXPIRED_TOKEN("만료된 토큰", 401),

    CUSTOMER_NOT_FOUND("해당 고객을 찾을 수 없음", 404),
    COUNSELLOR_NOT_FOUND("해당 상담사를 찾을 수 없음", 404),
    ADMIN_NOT_FOUND("해당 관리자를 찾을 수 없음", 404),
    OFFICIALS_NOT_FOUND("해당 관계자를 찾을 수 없음", 404),

    ALREADY_EXISTS_QUESTION("이미 존재하는 질문입니다.", 409),

    INTERNAL_SERVER_ERROR("서버 내부 에러", 500),
}